File: Java/RecyclerView/RecyclerView-mvvm/app/src/main/java/com/ls/project/recyclerview/adapter/ListRecyclerViewAdapter.java
package com.ls.project.recyclerview.adapter;

import android.databinding.ViewDataBinding;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.ls.project.recyclerview.BR;
import com.ls.project.recyclerview.databinding.ItemListBinding;
import com.ls.project.recyclerview.model.ListItem;

import java.util.List;

public class ListRecyclerViewAdapter extends RecyclerView.Adapter<ListRecyclerViewAdapter.GenericViewHolder> {
    List<ListItem> items;

    public ListRecyclerViewAdapter(List<ListItem> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public GenericViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        ItemListBinding binding = ItemListBinding.inflate(layoutInflater, parent,false);
        binding.linear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                LinearLayout linearLayout = (LinearLayout)v;

                Toast.makeText(v.getContext(),((TextView)((LinearLayout) v).getChildAt(0)).getText().toString() +" "+
                                ((TextView)((LinearLayout) v).getChildAt(1)).getText().toString()+" "+
                                ((TextView)((LinearLayout) v).getChildAt(2)).getText().toString()+" "+
                                ((TextView)((LinearLayout) v).getChildAt(3)).getText().toString()
                        ,Toast.LENGTH_SHORT).show();
            }
        });
        return new GenericViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull GenericViewHolder holder, int i) {
        holder.bind(items.get(i));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public void setItems(List<ListItem> items) {
        this.items = items;
    }

    public class GenericViewHolder extends RecyclerView.ViewHolder {
        private final ViewDataBinding binding;
        public GenericViewHolder(ItemListBinding binding) {
            super(binding.getRoot());
            this.binding=binding;
        }
        public void bind(ListItem item){
            binding.setVariable(BR.item,item);
            binding.executePendingBindings();
        }
    }
}

File: RecyclerView/RecycelerView-databinding/app/src/main/java/com/ls/project/areacard/adapter/ListRecyclerViewAdapter.kt
package com.ls.project.areacard.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.ls.project.areacard.BR
import com.ls.project.areacard.ListItem
import com.ls.project.areacard.databinding.ItemListBinding

class ListRecyclerViewAdapter(var items: List<ListItem>) :
    RecyclerView.Adapter<ListRecyclerViewAdapter.GenericViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, pos: Int): GenericViewHolder {
        var binding = ItemListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return GenericViewHolder(binding)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onBindViewHolder(holder: GenericViewHolder, pos: Int) {
        holder.bind(items[pos])
    }

    class GenericViewHolder(var binding: ItemListBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(item: ListItem) {
            binding.setVariable(BR.item, item)
            binding.executePendingBindings()
        }
    }
}